pkg/k8s/code/translate.go

```diff
@@ -0,0 +1,164 @@
+package code
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/okteto/okteto/pkg/k8s/secrets"
+	"github.com/okteto/okteto/pkg/model"
+
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	resource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	oktetoSyncLabel     = "syncthing.okteto.com"
+	syncImageTag        = "okteto/syncthing:0.4.2"
+	syncTCPPort         = 22000
+	syncGUIPort         = 8384
+	oktetoContainer     = "okteto"
+	oktetoSecretVolume  = "okteto-secret"
+	oktetoInitContainer = "okteto-init"
+	oktetoMount         = "/var/okteto"
+)
+
+var (
+	devReplicas int32 = 1
+)
+
+func translate(dev *model.Dev) *appsv1.StatefulSet {
+	initContainer := translateInitContainer(dev)
+
+	reqMem, _ := resource.ParseQuantity("64Mi")
+	reqCPU, _ := resource.ParseQuantity("50m")
+	limMem, _ := resource.ParseQuantity("256Mi")
+	limCPU, _ := resource.ParseQuantity("200m")
+	ss := &appsv1.StatefulSet{
```

Comment:
Why statefulspec vs deployment? (or even a pod directly)

Reply:
it is for pods that do now play well with restarts, and they usually mount volumes. it is a better fit for this case.